dask/array/core.py

```diff
@@ -1383,6 +1383,54 @@ def vindex(self):
         """
         return IndexCallable(self._vindex)
 
+    def _blocks(self, index):
+        from .slicing import normalize_index
+        if not isinstance(index, tuple):
+            index = (index,)
+        index = normalize_index(index, self.numblocks)
+        index = tuple(slice(k, k + 1) if isinstance(k, Number) else k
+                      for k in index)
+        name = 'getitem-' + tokenize(self, index)
```

Comment:
Would this clash with other `__getitem__` calls? Should we consider a different name here?

Reply:
I don't think it matters much in this case, but I've renamed this to `block`